x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/RequestXContent.java

```diff
@@ -216,11 +243,146 @@ private static QueryParams parseParams(XContentParser p) throws IOException {
                 )
             );
         }
-        if (errors.size() > 0) {
+        if (errors.isEmpty() == false) {
             throw new XContentParseException(
-                "Failed to parse params: " + String.join("; ", errors.stream().map(ex -> ex.getMessage()).toArray(String[]::new))
+                "Failed to parse params: "
+                    + String.join("; ", errors.stream().map(XContentParseException::getMessage).toArray(String[]::new))
             );
         }
         return new QueryParams(namedParams.isEmpty() ? unNamedParams : namedParams);
     }
+
+    private static void checkParamNameValidity(String name, List<XContentParseException> errors, XContentLocation loc) {
+        if (isValidParamName(name) == false) {
+            errors.add(
+                new XContentParseException(
+                    loc,
+                    "["
+                        + name
+                        + "] is not a valid parameter name, "
+                        + "a valid parameter name starts with a letter or underscore, "
+                        + "and contains letters, digits and underscores only"
+                )
+            );
+        }
+    }
+
+    private static ParamParsingKeys checkParamValueKeysValidity(
+        String paramKeyName,
+        List<XContentParseException> errors,
+        XContentLocation loc
+    ) {
+        ParamParsingKeys paramType = null;
+        try {
+            paramType = ParamParsingKeys.valueOf(paramKeyName.toUpperCase(Locale.ROOT));
+        } catch (IllegalArgumentException iae) {
+            errors.add(
+                new XContentParseException(
+                    loc,
+                    "["
+                        + paramKeyName
+                        + "] is not a valid param attribute, "
+                        + "a valid attribute is any of "
+                        + Arrays.stream(ParamParsingKeys.values())
+                            .map(p -> p.toString().toLowerCase(Locale.ROOT))
+                            .collect(Collectors.joining(", "))
+                )
+            );
+        }
+        return paramType;
+    }
```

Comment:
This is an anti-pattern since it:
1. ties the enum internal name to the external param
2. it makes the resolution throw an exception (not expensive but not cheap either).

Instead create a static map of String and associated enums to use for lookup:
```
private static final Map<String, ParamKey> paramTypeNames= Maps.newMapWithSize(ParamKey.values().size());

static {
   for (ParamKey e: ParamKey.values()) {
      paramKeyNames.put(e.name(), e);
  }
}

...
paramType = paramTypeNames.get(paramKeyName.toUpperCase(Locale.ROOT));
if (paramType == null) {
   // throw error
}
```

You can find this pattern through-out the code base in other places as well.